### PR TITLE
feat(server): support watch publicDir in dev

### PR DESCRIPTION
**Pull request by liyincode**
## Summary
support watch publicDir in dev
set `watch: true`, page will reload when watch file content changed.
```
    server: {
        publicDir: {
          watch: true,
        },
      },
``` 
## Related Links
#1903 

## Checklist

- [X] Tests updated (or not required).
- [X] Documentation updated (or not required).

### CODE REVIEW

File: examples/vue2/rsbuild.config.ts [hunk 1]
```diff
@@ -3,4 +3,10 @@ import { pluginVue2 } from '@rsbuild/plugin-vue2';
 
 export default defineConfig({
   plugins: [pluginVue2()],
+  server: {
+    publicDir: {
+      name: 'public1',
```
Comment (chenjiahan) [inline examples/vue2/rsbuild.config.ts:8]: is this a temp code😄

File: website/docs/en/config/server/public-dir.mdx [hunk 2]
```diff
@@ -63,6 +68,21 @@ export default {
...
+  server: {
+    publicDir: {
+      name: 'static',
+      watch: true,
+    },
+  },
+};
+```
+
+Note that the `watch` option is only valid in a development environment. If `dev.hmr` and `dev.liveReload` are both set to false, `watch` will be ignored.
```
Comment (chenjiahan) [inline website/docs/en/config/server/public-dir.mdx:84]:
```suggestion
Note that the `watch` option is only valid in the development build. If `dev.hmr` and `dev.liveReload` are both set to false, `watch` will be ignored.
```

File: examples/vue2/rsbuild.config.ts (hunk 1 above)
Comment (liyincode) [reply examples/vue2/rsbuild.config.ts]: 😅😅sorry, this is my debug code, i has removed.

File: website/docs/zh/config/server/public-dir.mdx [hunk 3]
```diff
@@ -16,6 +16,11 @@ type PublicDir =
        * @default true
        */
       copyOnBuild?: boolean;
+      /**
+       * 是否监控 public 目录 ，并在文件发生变化时重新加载页面
```
Comment (9aoy) [inline website/docs/zh/config/server/public-dir.mdx:20]:
```suggestion
       * 是否监听 public 目录，并在文件发生变化时重新加载页面
```

下面的 ‘监控’  也都改成 ‘监听’ 吧

Comment (liyincode) [reply website/docs/zh/config/server/public-dir.mdx]: 好的

**Review by chenjiahan — APPROVED** [review]

**Comment by liyincode** [comment]
> 🎉🎉

WoW🤟

**Pull request merged**